### PR TITLE
Feature/icp 18

### PR DESCRIPTION
**Pull request by OneTesseractInMultiverse**
- Implemented JWT Authentication Support using Nexo ID 1.0
- Implemented Authentication Service that allows checking if a user is authenticated. 
- Updated layout to include a conditional rendering of some components only if the user is logged in. 
- Created a login screen. 
- Implemented logout rules and router event subscription to update components state based on the router events.
- Packed all IAM logic in a portable Module. 
- Cleaned up some fuse components

### CODE REVIEW

File: src/app/layout/vertical/layout-1/layout-1.component.ts [hunk 1]
```diff
@@ -23,16 +28,29 @@ export class VerticalLayout1Component implements OnInit, OnDestroy
...
     )
     {
         // Set the defaults
         this.navigation = navigation;
 
         // Set the private defaults
         this._unsubscribeAll = new Subject();
+
+        router.events.subscribe((value => {
+            console.log(value);
```
Comment (Gabuardi) [inline src/app/layout/vertical/layout-1/layout-1.component.ts]: I think is a good idea to remove that `console logs` for the end user.

File: src/app/layout/vertical/layout-1/layout-1.component.ts [hunk 2]
```diff
@@ -23,16 +28,29 @@ export class VerticalLayout1Component implements OnInit, OnDestroy
...
+        private authService: AuthenticationService,
+        private router: Router
     )
     {
         // Set the defaults
         this.navigation = navigation;
 
         // Set the private defaults
         this._unsubscribeAll = new Subject();
+
```
Comment (Gabuardi) [inline src/app/layout/vertical/layout-1/layout-1.component.ts]: maybe add a method purpose comment?

File: src/app/layout/components/toolbar/toolbar.component.ts [hunk 3]
```diff
@@ -28,66 +31,42 @@ export class ToolbarComponent implements OnInit, OnDestroy
...
-            }
-        ];
 
         this.navigation = navigation;
 
         // Set the private defaults
         this._unsubscribeAll = new Subject();
+
+        router.events.subscribe((value => {
+            console.log(value);
```
Comment (Gabuardi) [inline src/app/layout/components/toolbar/toolbar.component.ts]: I think is a good idea to remove that console logs for the end user.

File: src/app/layout/vertical/layout-1/layout-1.component.ts (hunk 2 above)
Comment (OneTesseractInMultiverse) [reply src/app/layout/vertical/layout-1/layout-1.component.ts]: Will do

File: src/app/layout/components/toolbar/toolbar.component.ts (hunk 3 above)
Comment (OneTesseractInMultiverse) [reply src/app/layout/components/toolbar/toolbar.component.ts]: Yes, I'm going to note with a TODO and remove them when preparing for Test release. Now, I'll leave them just to debug potential issues.

**Pull request merged**